### PR TITLE
bump php requirement to 7.3

### PR DESCRIPTION
**Pull request by samsonasik**
I talked to @froschdesign about updating php requirement to php 7.3 today.

|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by samsonasik** [comment]
I removed one test: `test/unit/Adapter/Driver/Pgsql/ConnectionTest::testDisconnect()` as it requires `pg_close()` to pass resource and it cannot be mocked with pass null parameter.

**Comment by samsonasik** [comment]
@froschdesign travis build green :tada:

**Comment by samsonasik** [comment]
Ok, time for enable cs check (laminas coding standard v2) and coverage.

**Comment by samsonasik** [comment]
@weierophinney I've reverted last 3 commits.

**Comment by weierophinney** [comment]
> @weierophinney I've truncated last 3 commits.

That's not what I meant: I meant truncate the patch to ONLY THE ORIGINAL THREE commits (I should have been more clear). From there, see what changes you want to introduce to adopt PHP 7.3 features in a way that keeps backwards compatibility (e.g., typehints on private members, etc.)

**Comment by samsonasik** [comment]
@weierophinney updating components to use php 7.3 compat require unit tests changes as phpunit requirement changed. Truncating that will need to jump-cherry-pick-rebase between commits. I need to have backup branch, reset to make it 3 commit, and manually cherry-pick required commits will make it conflict changes. I will check if I have a chance to do it.

**Comment by samsonasik** [comment]
Ok, let me start over, start with update php req to 7.3 in composer.json and travis config.

**Comment by samsonasik** [comment]
next, updated with latest compatible component of php 7.3 including added phpspec/prophecy-phpunit.

**Comment by samsonasik** [comment]
update with latest phpunit `9.2` seems cause error in phpunit:

```bash
1) LaminasTest\Db\Adapter\AdapterTest::testQueryWhenPreparedWithParameterArrayProducesResult
Non unique mocked method invocation: Laminas\Db\Adapter\Driver\DriverInterface::createStatement

/Users/samsonasik/www/laminas-contribute/laminas-db/src/Adapter/Adapter.php:191
/Users/samsonasik/www/laminas-contribute/laminas-db/test/unit/Adapter/AdapterTest.php:245

2) LaminasTest\Db\Adapter\AdapterTest::testQueryWhenPreparedWithParameterContainerProducesResult
Non unique mocked method invocation: Laminas\Db\Adapter\Driver\DriverInterface::createStatement

/Users/samsonasik/www/laminas-contribute/laminas-db/src/Adapter/Adapter.php:191
/Users/samsonasik/www/laminas-contribute/laminas-db/test/unit/Adapter/AdapterTest.php:263
```

Need to pin as `~9.1.5`

**Comment by samsonasik** [comment]
@weierophinney Ok, I've updated with latest `~9.1.5` syntax. Can't update to php `9.2` as caused the previous error I mentioned.

the commit is now trimmed to only 3 commits:

- bump php requirement to 7.3 
- update components dependencies to use latest compatible 7.3
- update compatible latest phpunit 9 syntaxes

**Comment by samsonasik** [comment]
Travis build green :tada:

**Comment by samsonasik** [comment]
next is add `declare(strict_types=1)` ? is that ok?

File: .travis.yml [hunk 1]
```diff
@@ -12,49 +12,18 @@ env:
...
-    - php: 7.1
-      env:
-        - DEPS=lowest
-    - php: 7.1
-      services:
-        - mysql
-      env:
-        - DEPS=latest
-        - CS_CHECK=true
-        - TEST_COVERAGE=true
```
Comment (michalbundyra) [inline .travis.yml:44]: We need to move these to run CS CHECK and TEST COVERAGE on one of the supported PHP versions.

Comment (samsonasik) [reply .travis.yml:44]: I `did` that previously, but need to revert it based on @weierophinney comment at https://github.com/laminas/laminas-db/pull/164#pullrequestreview-424476322

Comment (samsonasik) [reply .travis.yml:44]: applied

**Comment by samsonasik** [comment]
I reverted back to use laminas coding standard to v1 and apply cs and test coverage in php 7.3 latest

**Comment by samsonasik** [comment]
Travis build green 🎉

**Comment by samsonasik** [comment]
merge-able?

**Comment by samsonasik** [comment]
Ok, I will create separate PR as alternative

**Comment by samsonasik** [comment]
@weierophinney work in progress alternative at #166

**Comment by michalbundyra** [comment]
Closing in favour of #166

**Pull request closed without merging**